test/libsolidity/SolidityABIJSON.cpp

```diff
@@ -756,7 +756,7 @@ BOOST_AUTO_TEST_CASE(library_function)
 	char const* sourceCode = R"(
 		library test {
 			struct StructType { uint a; }
-			function f(StructType storage b, uint[] storage c, test d) public returns (uint[] memory e, StructType storage f) {}
+			function f(StructType storage b, uint[] storage c, test d) public returns (uint[] memory e, StructType storage f) { f = f; }
```

Comment:
To be honest, I think this should also be an error - but you are saying the component is not thorough enough yet?

Reply:
There is cases, in which the component has no chance of being right, e.g.:

```
function f() retuns (S storage s) {
  if (true) { return f; }
}
```

This is of course a silly case, but in more complex scenarios it may make sense not to initialize in some code paths, because they will never be reached, but that fact cannot be determined without e.g. an SMT solver. That's why I think having this makes sense as a workaround.

Reply:
We also still don't check for cases like:
```
S m;
function f() returns (S storage s) {
   s.some_member = 4;
   s = m;
   return m;
}
```
Maybe we should have a separate issue for thinking about whether to do something about that...

Reply:
Less silly example for the question above:
```
function f(bool initializeEarly) public returns (S storage s) {
  uint valueNeededEarly;
  if(initializeEarly) (s, valueNeededEarly) = someThirdPartyFunction();
  else valueNeededEarly = 42;
  ...
  if(!initializeEarly) s = someOtherThirdPartyFunction(valueDependingOnValueNeededEarly);
}
```

Reply:
Arguably, this is no different than disallowing declaring ``S storage s;``, though, which we do disallow, I'll concede that...